src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -0,0 +1,65 @@
+using D2L.Bmx.Aws;
+using D2L.Bmx.Okta;
+
+namespace D2L.Bmx;
+
+internal class AwsCredsCreator {
+	private readonly IAwsClient _awsClient;
+	private readonly IConsolePrompter _consolePrompter;
+	private readonly BmxConfig _config;
+
+	public AwsCredsCreator(
+		IAwsClient awsClient,
+		IConsolePrompter consolePrompter,
+		BmxConfig config
+	) {
+		_awsClient = awsClient;
+		_consolePrompter = consolePrompter;
+		_config = config;
+	}
+
+	public async Task<AwsCredentials> CreateAwsCredsAsync(
+		IOktaApi oktaApi,
+		string? account,
+		string? role,
+		int? duration,
+		bool nonInteractive
+	) {
+		var accountState = await oktaApi.GetAccountsAsync( "amazon_aws" );
```

Comment:
In the future should we just hardcode this deeper in the API, in the context of BMX no other app is ever useful to filter for